tests/functional/test_snap.py

```diff
@@ -1,5 +1,157 @@
+import json
 import subprocess
+import urllib.request
+from contextlib import contextmanager
 
+from tenacity import Retrying, retry, stop_after_delay, wait_fixed
 
-def test_snap_installed():
-    subprocess.check_call(["snap", "list", "smartctl-exporter"])
+ENDPOINT = "http://localhost:9633/metrics"
+SYSTEMD_SERVICE = "snap.smartctl-exporter.smartctl-exporter"
+SNAP_NAME = "smartctl-exporter"
+
+
+@retry(wait=wait_fixed(2), stop=stop_after_delay(10))
+def _check_service_failed() -> None:
+    """Check if a systemd service is in a failed state."""
+    assert 0 == subprocess.call(
+        f"sudo systemctl is-failed --quiet {SYSTEMD_SERVICE}".split()
+    ), f"{SYSTEMD_SERVICE} is running"
+
+
+@retry(wait=wait_fixed(2), stop=stop_after_delay(10))
+def _check_service_active() -> None:
+    """Check if a systemd service is in a activemstate."""
+    assert 0 == subprocess.call(
+        f"sudo systemctl is-active --quiet {SYSTEMD_SERVICE}".split()
+    ), f"{SYSTEMD_SERVICE} is not running"
+
+
+@retry(wait=wait_fixed(5), stop=stop_after_delay(30))
+def _check_endpoint(endpoint: str) -> None:
```

Comment:
This suggestion is probably too wordy, but something a bit more descriptive would be helpful:

```suggestion
def _assert_endpoint_returns_success_response(endpoint: str) -> None:
```